components/BaseButton.vue

```diff
@@ -0,0 +1,47 @@
+<script setup lang="ts">
+import type { PropType } from 'vue';
+import { computed } from 'vue';
+import { NuxtLink } from '#components';
+
+const props = defineProps({
+  to: {
+    type: String,
+    default: '',
+  },
+  href: {
+    type: String,
+    default: '',
+  },
+  variant: {
+    type: String as PropType<'primary' | 'outline' | 'ghost'>,
+    default: 'primary',
+    validator: (v: string) => ['primary', 'outline', 'ghost'].includes(v),
+  },
+});
+
+const tag = computed(() => (props.to ? NuxtLink : props.href ? 'a' : 'button'));
+
+const baseBtn =
+  'inline-flex min-w-[176px] lg:min-w-[200px] justify-center py-[14px] rounded-[8px] border border-solid border-grey-100 ';
```

Comment:
most of [] values can be replaced by tailwind values. And if not as discussed let's use closest value.